### DISCUSSION TITLE
I can not train my own dataset on yolov8!

### DISCUSSION THREAD

**Discussion by 1nyourlife** [ultralytics/ultralytics#2938] [Q&A]
After update the ultralytics, I can not use the yaml files to train on my own dataset!

Before I update my ultralytics, I had write a new yaml file and put it in ../ultralytics/ultralytics/datasets, it worked well last week. But today after I input 'pip install -U ultralytics', it can not find my new yaml file like this error:
RuntimeError: Dataset 'data.yaml' error ❌ 'data.yaml' does not exist.

Besides, after I delete all files in ../ultralytics/ultralytics/datasets, it can still find the 'coco128.yaml' and begin training, even I delete all of the yaml files under this folder!

**Comment by glenn-jocher** [comment]
@1nyourlife hello! It seems like you're encountering an issue with locating your custom dataset YAML file after updating the Ultralytics package. Let's address this step by step:

1. **Custom YAML Location**: After updating, ensure that your custom YAML file is placed in the correct directory. The YAML file should be in the same directory where you are running your training command, or you should provide the correct relative path to the YAML file in the training command.

2. **YAML File Reference**: When you specify the dataset in your training command, make sure to reference the YAML file by its correct path. If it's in the current directory, just use the filename. If it's elsewhere, provide the relative or absolute path.

3. **Default Datasets**: The reason you can still train on `coco128.yaml` even after deleting YAML files from the datasets directory is that Ultralytics YOLOv8 automatically downloads default datasets like COCO128 if they are not found locally. This is a convenience feature to help users get started quickly.

4. **Dataset Cache**: YOLOv8 may cache dataset information. If you've moved or deleted files, you might need to clear this cache. Look for a file named `*.cache` (where `*` is a placeholder for your dataset name) and delete it to clear the cached data.

5. **Check for Typos**: Double-check the name of your YAML file and the reference in your training command for any typos.

If you continue to face issues, please ensure that your custom dataset is correctly formatted according to the documentation and that the paths in your YAML file are correct. If problems persist, you can refer to the documentation on how to train a model with a custom dataset for further guidance.
